### PR TITLE
Fix(49702): LS crash in binder for module in a JS file

### PR DESCRIPTION
**Pull request by navya9singh**
Fixes https://github.com/microsoft/TypeScript/issues/49702
Crash occurs due to unexpected `module` in a .js file. Adding a case in `bindThisPropertyAssignment()` to handle a `ModuleDeclaration` fixes the problem.

### CODE REVIEW

File: tests/cases/compiler/moduleCrashInJSFile.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//// [thisKeyword.ts]
+module foo {
+    this.bar = 4;
+}
+
+//// [thisKeyword.js]
+var foo;
+(function (foo) {
+    this.bar = 4;
+})(foo || (foo = {}));
```
Comment (sandersn) [inline tests/cases/compiler/moduleCrashInJSFile.ts:14]: This is part of the .js emit for the `thisKeyword` test and shouldn't be part of the test case.

File: tests/cases/compiler/moduleCrashInJSFile.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// @filename: a.js
+// @checkJs: true
+// @outDir: out/
+
+//// [thisKeyword.ts]
```
Comment (sandersn) [inline tests/cases/compiler/moduleCrashInJSFile.ts:5]: remove this line, because it hides the fact that this test is in a .js file.

**Review by sandersn — CHANGES_REQUESTED** [review]
Couple of small requests:
- add a comment
- clean up test

File: tests/cases/compiler/moduleCrashInJSFile.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// @filename: a.js
```
Comment (DanielRosenwasser) [inline tests/cases/compiler/moduleCrashInJSFile.ts:1]: Move the `filename` comment closer to the actual code. Then add another section with `// @filename: b.js` that has the same code but uses a `namespace` declaration.

Also, rename the containing file to `thisAssignmentInNamespaceDeclaration1.ts`.

Naming the test after the original behavior ("crash") is not really helpful long-term. Just name them after what's happening in the test, make sure we don't crash, and the baseline will demonstrate the actual behavior.

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
@DanielRosenwasser do you want to take another look at this? Because I think it's ready to merge now.

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**